lib/util/src/platform.c

```diff
@@ -25,12 +25,61 @@
 
 unsigned long hse_tsc_freq HSE_READ_MOSTLY;
 unsigned int hse_tsc_mult HSE_READ_MOSTLY;
-unsigned int hse_tsc_shift HSE_READ_MOSTLY;
 
 const char *hse_progname HSE_READ_MOSTLY;
 
+thread_local volatile const char *hse_wmesg_tls;
+
+extern rest_get_t workqueue_rest_get;
 extern rest_get_t kmc_rest_get;
 
+/* usleep(3) is simple to use but made obsolete by the more cumbersome
+ * nanosleep(2).  So we implement our own version built upon nanosleep
+ * to leverage its well-known and reliable semantics.
+ */
+int
+usleep(useconds_t usec)
+{
+    struct timespec req;
+
+    req.tv_nsec = (usec % USEC_PER_SEC) * 1000;
+    req.tv_sec = usec / USEC_PER_SEC;
+
+    return hse_nanosleep(&req, NULL, "usleep");
+}
+
+ssize_t
+hse_readfile(int dirfd, const char *path, void *buf, size_t bufsz, int flags)
```

Comment:
How often are we doing this that it requires its own function?

Reply:
There are currently three use cases of which hse_meminfo() is called more than I would like, and the workqueue rest hook will probably see quite a bit of use.  I have plans for other use cases.  I suspect we should use it in cgroup_name_get() as well...

Reply:
maybe this should just be made a static function? Or it belongs in a better header file than platform.h

Reply:
If it was static then no one could call it???
If we expect that one day soon Linux will grow a readfile(2) system call then it will definitely be a platform concern.
The "platform" problem is that tree isn't organized correctly to support more than one platform, hence we have general and linux-specific platform code all over the place.  So, not sure where you would put this if not under "platform" ???

Reply:
hse_util/unistd.h maybe...